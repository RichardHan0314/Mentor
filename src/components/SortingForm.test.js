import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import SortingForm from './SortingForm';

describe('SortingForm component', () => {
    it('Render SortingForm snapshot', () => {
        const sortingOnChange = jest.fn();
        const { asFragment } = render(<SortingForm sortingOnChange={sortingOnChange}/>);
        expect(asFragment(<SortingForm />)).toMatchSnapshot();
    });

    it('calls the onChange callback handler', () => {
        const sortingOnChange = jest.fn();

        render(
            <SortingForm sortingOnChange={sortingOnChange}/>
        );

        const select = screen.getByRole('combobox');
        fireEvent.change(select);
        expect(sortingOnChange).toHaveBeenCalledTimes(1);
    });
});