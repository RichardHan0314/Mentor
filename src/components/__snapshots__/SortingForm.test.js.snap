// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingForm component Render SortingForm snapshot 1`] = `
<DocumentFragment>
  <form
    class="sc-AxjAm iEUOGh filter-menu"
  >
    <label
      class="sc-AxirZ khLcSR"
      for="seriesInput"
    >
      Sort By
    </label>
    <select
      class="sc-AxiKw bczocQ"
      id="seriesInput"
    >
      <option
        value="release_date"
      >
        release date
      </option>
      <option
        value="title"
      >
        movie name
      </option>
      <option
        value="genres"
      >
        genre
      </option>
    </select>
  </form>
</DocumentFragment>
`;
